src/transformers/models/patchtsmixer/modeling_patchtsmixer.py

```diff
@@ -0,0 +1,2362 @@
+            Defaults to "common_channel".
+
+    """
+
+    def __init__(
+        self,
+        num_features: int = 16,
+        expansion_factor: int = 2,
+        dropout: float = 0.2,
+        gated_attn: bool = False,
+        ffn: str = "mlp",
+        mode: str = "common_channel",
+        norm_mlp: str = "LayerNorm",
+    ):
+        super().__init__()
+        self.norm_mlp = norm_mlp
+        self.mode = mode
+        self.norm = PatchTSMixerNormLayer(norm_mlp=norm_mlp, mode=mode, num_features=num_features)
+
+        self.mlp = PatchTSMixerMLP(num_features, num_features, expansion_factor, dropout)
+
+        self.gated_attn = gated_attn
+
+        if self.gated_attn:
+            self.gab = PatchTSMixerGatedAttention(in_size=num_features, out_size=num_features)
+
+    def forward(self, x):
+        # x.shape == (batch_size, num_patches, num_features) if flatten
+        # x.shape == (batch_size, n_vars, num_patches, num_features) if common_channel
+
+        residual = x
+
+        x = self.norm(x)
+
+        # if self.gated_attn:
+        #     x = self.gab(x)
+
+        x = self.mlp(x)
+        # x.shape == (batch_size, num_patches, num_features) if flatten
+        # x.shape == (batch_size, n_vars, num_patches, num_features) if common_channel
+
+        if self.gated_attn:
+            x = self.gab(x)
+
+        out = x + residual
+        return out
+
+
+class PatchTSMixerLayer(nn.Module):
+    """
+    Args:
+        num_features (`int`, *optional*): Hidden feature size. Defaults to 16.
+        num_patches (`int`): Number of patches to segment
+        in_channels (`int`, *optional*): Number of input variables. Defaults to 3.
+        expansion_factor (`int`, *optional*): Expansion factor to use inside MLP. Defaults to 2.
+        dropout (`float`, *optional*): Backbone Dropout rate. Defaults to 0.2.
+        mode (str, *optional*): Mixer Mode. Determines how to process the channels. Allowed values: flatten,
+            common_channel, mix_channel. In flatten, patch embedding encodes the patch information across all channels.
+            In common_channel mode, patch embedding is independent of channels (Channel Independece). In mix_channel,
+            we follow channel independence, but in addition to patch and feature mixing, we also do channel mixing.
+            Defaults to "common_channel".
+        gated_attn (bool, *optional*): Enable Gated Attention. Defaults to False.
+        ffn (str, *optional*): MLP mode. Allowed values: mlp, gmlp. gmlp is not preferred. Defaults to "mlp".
+        self_attn (bool, *optional*): Enable Tiny self attention in addition to MLP mixing. Defaults to False.
+        self_attn_heads (bool, *optional*): Self attention heads. Defaults to 1.
+        norm_mlp (str, *optional*): Norm layer (BatchNorm or LayerNorm). Defaults to LayerNorm.
+
+    """
+
+    def __init__(
+        self,
+        num_patches: int,
```

Comment:
typically our classes just take the `config` as input, rather than all individual args.

Not sure if possible here?

Reply:
Hi @NielsRogge , thanks for the comment. These are some classes that are used internally. The main exposed classes which are generally inherited from `PatchTSMixerPreTrainedModel`, take `config` as the input. Is this okay?

Reply:
We usually also do this for internal classes as well

Reply:
Sure @NielsRogge We enabled this change to all the classes that get initialized with the config params. We now pass it as config, instead of passing every param individually. 

@ajati - this requirement had a lot of code refactoring. Please review it carefully from your side too.

Reply:
Sure.